core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/protocol/json/SimpleTypeJsonMarshaller.java

```diff
@@ -89,10 +94,34 @@ public void marshall(Boolean val, StructuredJsonGenerator jsonGenerator, JsonMar
         }
     };
 
-    public static final JsonMarshaller<Instant> INSTANT = new BaseJsonMarshaller<Instant>() {
+    public static final JsonMarshaller<Instant> INSTANT = new JsonMarshaller<Instant>() {
         @Override
-        public void marshall(Instant val, StructuredJsonGenerator jsonGenerator, JsonMarshallerContext context) {
-            jsonGenerator.writeValue(val);
+        public void marshall(Instant val, JsonMarshallerContext context, String paramName, SdkField<Instant> sdkField) {
+            StructuredJsonGenerator jsonGenerator = context.jsonGenerator();
+            if (paramName != null) {
+                jsonGenerator.writeFieldName(paramName);
+            }
+            TimestampFormatTrait trait = sdkField.getTrait(TimestampFormatTrait.class);
+            if (trait != null) {
```

Comment:
Similar logic is present in ValueToStringConverter.FROM_INSTANT. Can you use it here?

Reply:
They differ subtly in such a way that makes it hard to combine. For example we are always producing a String in ValueToString but we need to know how to write it to the JSON generator (i.e. as a string or as a string number). And we also have the fallback which is needed for CBOR.